RIOT-OS/RIOT
# core: implement core_mutex_priority_inheritance

### Contribution description

This implements `core_mutex_mitigate_priority_inversion`, a module that employs the textbook solution of priority inheritance to mitigate priority inversion.

In addition, the corresponding test was cleaned up and automated:
- drop timer dependency, just use mutex to stage the priority inversion scenario
- reduce memory consumption so that `Makefile.ci` can be dropped by using `fmt` over stdio and reducing stack sizes
- reduce the complexity of the scenario to the bare minimum required to produce the issue

### Testing procedure

```
make flash test -C tests/thread_priority_inversion
```

should build, flash and run. It should fail without `core_mutex_mitigate_priority_inversion` (comment out line in the test's `Makefile`), but pass with if it is used.

### Issues/PRs references

None
